src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -1061,14 +1073,62 @@ public void OnSearchElementClicked(NodeModel nodeModel, Point position)
 
         internal void OnRequestConnectToPort(string nodeCreationName, PortModel portModel)
         {
-            if (!nodeCreationName.Contains("Code Block"))
+            // Do not auto connect code block node since default code block node do not have output port
+            if (nodeCreationName.Contains("Code Block")) return;
+            
+            var initialNode = dynamoViewModel.CurrentSpaceViewModel.Nodes.FirstOrDefault(x => x.Id == portModel.Owner.GUID);
+            var id = Guid.NewGuid();
+
+            var adjustedX = initialNode.X;
+
+            // Placing the new node based on which input port it is connecting to.
+            if (portModel.PortType == PortType.Input)
             {
-                // Create a new node based on node creation name and connect ports
-                dynamoViewModel.ExecuteCommand(new DynamoModel.CreateAndConnectNodeCommand(Guid.NewGuid(), portModel.Owner.GUID,
-                    nodeCreationName, 0, portModel.Index, portModel.CenterX - 50, portModel.CenterY + 200, false, false));
+                // Placing the new node to the left of initial node
+                adjustedX -= portModel.Owner.Width + 50;
+            }
+            else
+            {
+                // Placing the new node to the right of initial node
+                adjustedX += portModel.Owner.Width + 50;
             }
 
-            OnRequestFocusSearch();
+            // Initialize a new undo action group before calling 
+            // node CreateAndConnect and AutoLayout commands.
+            if (undoRecorderGroup == null)
+            {
+                // Node auto layout can be performed correctly only when the positions and sizes
+                // of nodes are known, which is possible only after the node views are ready.
+                dynamoViewModel.NodeViewReady += AutoLayoutNodes;
+                undoRecorderGroup = dynamoViewModel.CurrentSpace.UndoRecorder.BeginActionGroup();
+            }
+
+            // Create a new node based on node creation name and connection ports
+            dynamoViewModel.ExecuteCommand(new DynamoModel.CreateAndConnectNodeCommand(id, portModel.Owner.GUID,
+                nodeCreationName, 0, portModel.Index, adjustedX, 0, false, false, true));
+
+            // Clear current selections and select all input nodes as we need to perform Auto layout on only the input nodes.
+            DynamoSelection.Instance.ClearSelection();
+            var inputNodes = portModel.Owner.InputNodes.Values.Where(x => x != null).Select(y => y.Item2);
+
+            foreach (var inputNode in inputNodes)
+            {
+                DynamoSelection.Instance.Selection.AddUnique(inputNode);
```

Comment:
Oh @aparajit-pratap I see, you are touching the `Selection` directly rather than calling `Select()` on each inidividual node. That explains why my approach earlier may not work.. Good that you have a fresh eye on this!